api/src/main/java/org/openmrs/util/DatabaseUpdater.java

```diff
@@ -359,36 +404,36 @@ private static Liquibase getLiquibase(String changeLogFile, ClassLoader cl) thro
 		}
 		catch (SQLException e) {
 			throw new Exception(
-			        "Unable to get a connection to the database.  Please check your openmrs runtime properties file and make sure you have the correct connection.username and connection.password set",
-			        e);
+				"Unable to get a connection to the database.  Please check your openmrs runtime properties file and make sure you have the correct connection.username and connection.password set",
+				e);
 		}
-		
+
 		if (cl == null) {
 			cl = OpenmrsClassLoader.getInstance();
 		}
-		
+
 		try {
 			Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(
-			    new JdbcConnection(connection));
+				new JdbcConnection(connection));
 			database.setDatabaseChangeLogTableName("liquibasechangelog");
 			database.setDatabaseChangeLogLockTableName("liquibasechangeloglock");
-			
+
 			if (connection.getMetaData().getDatabaseProductName().contains("HSQL Database Engine")
-			        || connection.getMetaData().getDatabaseProductName().contains("H2")) {
+				|| connection.getMetaData().getDatabaseProductName().contains("H2")) {
 				// a hack because hsqldb and h2 seem to be checking table names in the metadata section case sensitively
 				database.setDatabaseChangeLogTableName(database.getDatabaseChangeLogTableName().toUpperCase());
 				database.setDatabaseChangeLogLockTableName(database.getDatabaseChangeLogLockTableName().toUpperCase());
 			}
-			
+
 			ResourceAccessor openmrsFO = new ClassLoaderFileOpener(cl);
 			ResourceAccessor fsFO = new FileSystemResourceAccessor();
-			
+
 			if (changeLogFile == null) {
-				changeLogFile = CHANGE_LOG_FILE;
+				changeLogFile = EMPTY_CHANGE_LOG_FILE;
 			}
-			
-			database.checkDatabaseChangeLogTable(false, null, null);
-			
+
+			ChangeLogHistoryServiceFactory.getInstance().getChangeLogService(database).init();
```

Comment:
This is a tough one, I need to think hard about it. It is related to upgrading Liquibase from 2.0.5 to 3.6.1 (and later to 3.8.9). There were breaking changes between 2.0.5 and 3.x which I accommodated a long time ago. One thing I realised is that before squashing all the commits for the review I should create a backup branch with the original commit history so that I can trace changes back better. Well, next time I suppose, we are are here to learn. I shall look into that over the weekend.

Reply:
So, I did some research. The change happened in the context of upgrading Liquibase from 2.x to 3.y. There were a few breaking changes and I had replaced deprecated things with respective calls that are based on the current Liquibase API.

I tried initialisation of OpenMRS in different ways:
(a) using TRUNK-4830-N and without the line you commented on
(b) using master and without the previous version of that line

Interestingly both initialisations were successful. I am still hesitant to remove that line as giving it a try once is not sufficient to be sure that this call is not needed at all.

Unfortunately the Liquibase documentation is rather silent, the decompiled class does not contain any Javascript.

So I am inclined to leave that line of code in place, following a "better safe than sorry" approach. This is not satisfying but retrofitting this and other classes with all the missing tests would be a story (if not epic) in its own right.

Reply:
Please let me know what you think, Ian as this is the last outstanding comment.

Reply:
I'd just add a comment like:

```java
// ensure the changelog service is initialised
```

I just want to be sure someone doesn't come along and remove it. Thanks for the digging on this. And it does look like [the method can safely be called multiple times](https://github.com/liquibase/liquibase/blob/a930d0241b03602ef1bee30d2bd725bcd77b1665/liquibase-core/src/main/java/liquibase/changelog/StandardChangeLogHistoryService.java#L99).

Reply:
thanks, will do, have a sunny weekend!